go/service/gregor.go

```diff
@@ -521,7 +523,33 @@ func (g *gregorHandler) iterateOverFirehoseHandlers(f func(h libkb.GregorFirehos
 	g.firehoseHandlers = freshHandlers
 }
 
-func (g *gregorHandler) pushState(r keybase1.PushReason) {
+func (g *gregorHandler) pushStateThread(shutdownCh chan struct{}) {
+	shouldSend := false
+	var lastTime time.Time
+	dur := time.Second
+	trigger := func() {
+		if shouldSend {
+			go g.pushStateOnce(keybase1.PushReason_NEW_DATA)
+			shouldSend = false
+			lastTime = time.Now()
+		}
+	}
+	for {
+		select {
+		case <-g.pushStateCh:
+			shouldSend = true
+			if time.Since(lastTime) > dur {
+				trigger()
+			}
+		case <-time.After(dur):
+			trigger()
+		case <-shutdownCh:
```

Comment:
this never gets reinitialized on reconnect

Reply:
It does in `Connect`

Reply:
oops missed this, lgtm!